scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,559 @@
+            args : tuple
+                Extra arguments passed to the objective function (``func``) and
+                its derivatives (Jacobian, Hessian).
+    initial_temp : float, optional
+        The initial temperature, use higher values to facilitates a wider
+        search of the energy landscape, allowing simulated_annealing  to escape
+        local minima that it is trapped in.
+    visit : float, optional
+        Parameter for visiting distribution. Higher values give the visiting
+        distribution a heavier tail, this makes the algorithm jump to a more
+        distant region. The value range is (0, 3]
+    accept : float, optional
+        Parameter for acceptance distribution. It is used to control the
+        probability of acceptance. The lower the acceptance parameter, the
+        smaller the probability of acceptance. It has to be any negative value.
+    maxfun : int, optional
+        Soft limit for the number of objective function calls. If the
+        algorithm is in the middle of a local search, this number will be
+        exceeded, the algorithm will stop just after the local search is
+        done.
+    seed : int or `np.random.RandomState`, optional
+        If `seed` is not specified the `np.RandomState` singleton is used.
+        If `seed` is an int, a new `np.random.RandomState` instance is used,
+        seeded with seed.
+        If `seed` is already a `np.random.RandomState instance`, then that
+        `np.random.RandomState` instance is used.
+        Specify `seed` for repeatable minimizations. The random numbers
+        generated with this seed only affect the visiting distribution
+        function and new coordinates generation.
+    no_local_search : boolean, optional
+        If `no_local_search` is set to `True`, a traditional Generalized
+        Simulated Annealing will be performed with no local search
+        strategy applied.
+
+    Returns
+    -------
+    res : OptimizeResult
+        The optimization result represented as a ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``fun`` the value
+        of the function at the solution, and ``message`` which describes the
+        cause of the termination.
+        See `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+    SDA is an implementation of the Simulated Dual Annealing. This stochastic
+    approach [3]_ implements the generalization of CSA (Classical Simulated
+    Annealing) and FSA (Fast Simulated Annealing) [1]_ [2]_ coupled to a
+    strategy for applying a local search on accepted locations [4]_. A first
+    implementation was done in C++ for the R language [5]_ and has been
+    benchmarked [6]_. Extensive benchmarks for the Python version have also
+    been performed against several global optimization methods [7]_. SDA
+    introduces an advanced dual annealing method to refine the solution found
```

Comment:
I do not like the phrase:

``A first  implementation was done in C++ for the R language [5]_ and has been benchmarked [6]_. Extensive benchmarks for the Python version have also  been performed against several global optimization methods [7]_.``

 it gives very few information that is actually useful to someone using your code. Maybe reduce it to something like:
``Alternative implementations of this same algorithm are described in [5]_  and benchmarks are presented in [6]_ and [7]_.``

Reply:
Adjusted.